core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -4504,63 +4504,64 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     assert(mapStatuses.count(s => s != null && s.location.executorId == "hostB-exec") === 1)
   }
 
-  Seq(true, false).foreach { registerMergeResults =>
-    test("SPARK-40096: Send finalize events even if shuffle merger blocks indefinitely " +
-      s"with registerMergeResults is ${registerMergeResults}") {
-      initPushBasedShuffleConfs(conf)
-
-      sc.conf.set("spark.shuffle.push.results.timeout", "1s")
-      val myScheduler = new MyDAGScheduler(
-        sc,
-        taskScheduler,
-        sc.listenerBus,
-        mapOutputTracker,
-        blockManagerMaster,
-        sc.env,
-        shuffleMergeFinalize = false)
-
-      val mergerLocs = Seq(makeBlockManagerId("hostA"), makeBlockManagerId("hostB"))
-      val timeoutSecs = 1
-      val sendRequestsLatch = new CountDownLatch(mergerLocs.size)
-      val completeLatch = new CountDownLatch(mergerLocs.size)
-      val canSendRequestLatch = new CountDownLatch(1)
-
-      val blockStoreClient = mock(classOf[ExternalBlockStoreClient])
-      val blockStoreClientField = classOf[BlockManager].getDeclaredField("blockStoreClient")
-      blockStoreClientField.setAccessible(true)
-      blockStoreClientField.set(sc.env.blockManager, blockStoreClient)
-
-      val sentHosts = ArrayBuffer[String]()
-      var hostAInterrupted = false
-      doAnswer { (invoke: InvocationOnMock) =>
-        val host = invoke.getArgument[String](0)
-        sendRequestsLatch.countDown()
-        try {
-          if (host == "hostA") {
-            canSendRequestLatch.await(timeoutSecs * 2, TimeUnit.SECONDS)
-          }
-          sentHosts += host
-        } catch {
-          case _: InterruptedException => hostAInterrupted = true
-        } finally {
-          completeLatch.countDown()
+  test("SPARK-40096: Send finalize events even if shuffle merger blocks indefinitely ") {
+    initPushBasedShuffleConfs(conf)
+
+    sc.conf.set("spark.shuffle.push.results.timeout", "1s")
+    val scheduler = new DAGScheduler(
+      sc,
+      taskScheduler,
+      sc.listenerBus,
+      mapOutputTracker,
+      blockManagerMaster,
+      sc.env)
```

Comment:
Do we need the rest of the changes ? I assumed simply changing from `MyDAGScheduler` -> `DAGScheduler` was sufficient ?